.esbuild/plugins/qgds-plugin-generate-icon-assets.js

```diff
@@ -0,0 +1,103 @@
+import fs from "fs";
+import path from "path";
+import log from "../helpers/logger.js";
+import svgToMiniDataURI from "mini-svg-data-uri";
+import { parse } from "node-html-parser";
+
+const PREFIX_QGDS = 'qgds-icon-';
+
+/**
+ * Generate CSS variables of icons from SVG files.
+ * @param {string} inputDir - Directory containing SVG icon files
+ * @param {string} outputIconVars - Output SASS file for the generated CSS variables
+ * @param {string} outputIconNames - Output SASS file for the generated icon names - to build icon utility classes
+ * @param {string} outputIconJs - Output JS file for the generated icon names - to be used in Storybook
+ */
+export function QGDSgenerateIconAssetsPlugin(inputDir = "./src/img/icons", 
+  outputIconSprite = "./src/img/_icon-sprite.svg",
```

Comment:
this looks like it should be moved to its own repo as this code is agnostic to bs5.